### PR TITLE
Remove unnecessary _catch suffix from files.

### PR DESCRIPTION
**Pull request by rcurtin**
This just changes the name of the files `serialization_catch.hpp` and `serialization_catch.cpp` to `serialization.hpp` and `serialization.cpp`, respectively.  There's no need for a `_catch` suffix anymore, since the Boost -> catch transition is done now. :)

### CODE REVIEW

File: src/mlpack/tests/ann_layer_test-mod.cpp [hunk 1]
```diff
@@ -0,0 +1,2937 @@
+/**
+ * @file mlpack/tests/ann_layer_test-mod.cpp
```
Comment (jeffin143) [inline src/mlpack/tests/ann_layer_test-mod.cpp:2]: What is this change , I am not sure about this change

Comment (rcurtin) [reply src/mlpack/tests/ann_layer_test-mod.cpp]: Ahh, nice catch, that's something I didn't mean to check in!  Fixed in 700e5ee72.

**Comment by jeffin-ntx** [comment]
@mlpack-jenkins test this again

**Comment by jeffin143** [comment]
@mlpack-jenkins test this again

**Comment by zoq** [comment]
> @mlpack-jenkins test this again

Safe to ignore the Static Analysis issue.

**Review by zoq — APPROVED** [review]
Nice, no comments from my side.

**Review by jeffin143 — APPROVED** [review]
Let's close the issue :100:

**Pull request merged**